arguebuf/load/_load_aif.py

```diff
@@ -16,17 +17,230 @@
 __all__ = ("load_aif",)
 
 
+class NewNode:
+    def __init__(self, node_id, text, node_type, timestamp):
+        self.node_id = node_id
+        self.text = text
+        self.node_type = node_type
+        self.timestamp = timestamp
+        self.incoming_edges = []
+        self.outgoing_edges = []
+
+    def add_incoming_edge(self, edge):
+        self.incoming_edges.append(edge)
+
+    def add_outgoing_edge(self, edge):
+        self.outgoing_edges.append(edge)
+
+    def __repr__(self):
+        return f"Node(node_id={self.node_id}, text={self.text}, type={self.node_type}, timestamp={self.timestamp}, incoming_edges={self.incoming_edges}, outgoing_edges={self.outgoing_edges})"
+
+
+def get_connected_nodes_of_type(node, node_type, nodes):
+    """Returns a list of nodes connected to the given node (either incoming or outgoing) of a specific type."""
+    connected_node_ids = [edge['fromID'] for edge in node.incoming_edges] + [edge['toID'] for edge in
+                                                                             node.outgoing_edges]
+    return [nodes[node_id] for node_id in connected_node_ids if nodes[node_id].node_type == node_type]
+
+
+# Function to get connected nodes excluding certain types
+def get_connected_nodes_excluding_types(node, excluded_types, nodes):
+    """Returns a list of nodes connected to the given node (either incoming or outgoing) excluding specific types."""
+    connected_node_ids = [edge['fromID'] for edge in node.incoming_edges] + [edge['toID'] for edge in
+                                                                             node.outgoing_edges]
+    return [nodes[node_id] for node_id in connected_node_ids if nodes[node_id].node_type not in excluded_types]
+
+
+# Function to check if two nodes are connected
+def are_nodes_connected(node1, node2):
+    """Returns True if node1 and node2 are connected (either node1 to node2 or node2 to node1), otherwise False."""
+    return any(edge['fromID'] == node1.node_id and edge['toID'] == node2.node_id for edge in node1.outgoing_edges) or \
+        any(edge['fromID'] == node2.node_id and edge['toID'] == node1.node_id for edge in node2.outgoing_edges)
+
+
+# Function to generate a unique ID (this is a placeholder, replace with your ID generation logic)
+def generate_unique_id():
+    return str(uuid.uuid4())
+
+
+def remove_loops_in_hanging_nodes(hanging_nodes, obj, nodes):
+    # Check for and remove loops
+    # 1) For each hanging_node take any outgoing edge
+    # 2) For each outgoing edge that check what argumentNode it reaches, if its not in the list of hanging_nodes return
+    # If it was a hanging_node find the outgoing edge that goes back to the previous hanging_node and delete this path
+    """Remove loops formed by hanging nodes."""
```

Comment:
Please combine comments and docstrings into one "field" Use the following syntax:

```txt
"""Title

Additional
text
here
"""
```